packages/x-codemod/src/util/removeObjectProperty.ts

```diff
@@ -48,11 +48,19 @@ export default function removeObjectProperty({
         return;
       }
       const definedKeys: any[] = [];
-      const objectProperties = j(targetAttribute).find(j.Property);
-      objectProperties.forEach((path) => {
-        const objectKey = (path.value.key as any).name as any;
-        definedKeys.push(objectKey);
-      });
+      const properties = j(targetAttribute).find(j.Property);
+      const objectProperties = j(targetAttribute).find(j.ObjectProperty);
+
+      const propertiesToProcess = properties.length > 0 ? properties : objectProperties;
+      if (propertiesToProcess.length > 0) {
```

Comment:
if the length is 0, enclosed code will not do anything anyway

Reply:
Correct, I actually wanted to do https://github.com/mui/mui-x/pull/14150/commits/2befa5a1feea45fe80eb5cc24e6cf0cc8672df73 here. Updated!